aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -52,9 +51,29 @@ pub(crate) fn get_resource_group_from_metadata(
         .find_map(|attr| attr.get_resource_group_member())
 }
 
-/// Adapter to convert a `StateView` into a `MoveResolverExt`.
-pub struct StorageAdapter<'a, S> {
-    state_store: &'a S,
+// Allows to keep a single `StorageAdapter` for both borrowed or owned views.
+// For example, views are typically borrowed during block execution, but are
+// owned in tests or in indexer.
+enum ExecutorViewKind<'e, E: 'e> {
```

Comment:
Wanted to use it initially but the cloning part stopped me :/ because then we need to implement clone for ExecutorView, which doesn't seems nice? What do you think @gelash @zekun000

Reply:
You could implement https://doc.rust-lang.org/std/borrow/trait.ToOwned.html and leave it unimplemented!()
and then never call to_owned method 🗡️ 
Let's see what @zekun000 thinks is less ugly? xD

Reply:
Will leave like that for now, once @zekun000 has a verdict which approach is uglier will send another PR :)